cosi/cosi.go

```diff
@@ -1,450 +1,399 @@
-	// challenge holds the challenge for this round
-	challenge abstract.Scalar
-
-	// the longterm private key CoSi will use during the response phase.
-	// The private key must have its public version in the list of publics keys
-	// given to CoSi.
-	private abstract.Scalar
-	// random is our own secret that we wish to commit during the commitment phase.
-	random abstract.Scalar
-	// commitment is our own commitment
-	commitment abstract.Point
-	// response is our own computed response
-	response abstract.Scalar
-	// aggregateResponses is the aggregated response from the children + our own
-	aggregateResponse abstract.Scalar
-}
-
-// NewCosi returns a new Cosi struct given the suite, the longterm secret, and
-// the list of public keys. If some signers were not to be participating, you
-// have to set the mask using `SetMask` method. By default, all participants are
-// designated as participating. If you wish to specify which co-signers are
-// participating, use NewCosiWithMask
-func NewCosi(suite abstract.Suite, private abstract.Scalar, publics []abstract.Point) *CoSi {
-	cosi := &CoSi{
-		suite:   suite,
-		private: private,
-	}
-	// Start with an all-disabled participation mask, then set it correctly
-	cosi.mask = newMask(suite, publics)
-	return cosi
-}
-
-// CreateCommitment creates the commitment of a random secret generated from the
-// given s stream. It returns the message to pass up in the tree. This is
-// typically called by the leaves.
-func (c *CoSi) CreateCommitment(s cipher.Stream) abstract.Point {
-	c.genCommit(s)
-	return c.commitment
+// Commit returns a random scalar v, generated from the given cipher stream, and
+// a corresponding commitment V = [v]G.
+func Commit(suite abstract.Suite, s cipher.Stream) (abstract.Scalar, abstract.Point) {
+	var stream = s
+	if s == nil {
+		stream = random.Stream
+	}
+	random := suite.Scalar().Pick(stream)
+	commitment := suite.Point().Mul(nil, random)
+	return random, commitment
 }
 
-// Commit creates the commitment / secret as in CreateCommitment and it also
-// aggregate children commitments from the children's messages.
-func (c *CoSi) Commit(s cipher.Stream, subComms []abstract.Point) abstract.Point {
-	// generate our own commit
-	c.genCommit(s)
-
-	// add our own commitment to the aggregate commitment
-	c.aggregateCommitment = c.suite.Point().Add(c.suite.Point().Null(), c.commitment)
-	// take the children commitments
-	for _, com := range subComms {
-		c.aggregateCommitment.Add(c.aggregateCommitment, com)
+// AggregateCommitments returns the sum of the given commitments and the
+// bitwise OR of the corresponding masks.
+func AggregateCommitments(suite abstract.Suite, commitments []abstract.Point, masks [][]byte) (abstract.Point, []byte, error) {
+	if len(commitments) != len(masks) {
+		return nil, nil, errors.New("mismatching lengths of commitment and mask slices")
 	}
-	return c.aggregateCommitment
-
+	aggCom := suite.Point().Null()
+	aggMask := make([]byte, len(masks[0]))
```

Comment:
It is, see https://github.com/dedis/kyber/blob/3f73f33656c1c5e2ebca141ef09cfba321edb551/cosi/cosi.go#L76.